rcldotnet/Node.cs

```diff
@@ -375,6 +303,22 @@ private Subscription<T> CreateSubscriptionInner<T>(string topic, Action<T> callb
             return client;
         }
 
+        public Timer CreateTimer(TimeSpan period, Action<TimeSpan> callback)
+        {
+            Timer timer = new Timer(Clock, period, callback);
+            _timers.Add(timer);
+            return timer;
+        }
+
+        public bool DestroyTimer(Timer timer)
+        {
+            if (!_timers.Contains(timer)) return false;
+
+            _timers.Remove(timer);
+            timer.Handle.Dispose();
```

Comment:
Other objects like `Subscription` don't support manual dispose right now.
If I remember correctly they are not allowed to be deallocated while some thread is waiting for them on a `WaitSet`. This needs some more thought to get it right.
I would for now opt to not implement this method.
This is something for an extra PR to do this for the other "Waitables" as well.
I wouldn't implement Destroy methods, but let the Objects themself implement the `IDisposable` interface.

Reply:
Maybe we can make this method internal for now if we can prove that the usages for `use_sim_time` don't happen while the waitset is active. This is the case if this method only gets called in a callback that gets triggered when the waist has ready elements.

> For the objects themselves to handle their own disposal, I suppose they would need a reference to the Node they are associated with to request that the Node run their Dispose at the correct time and remove their own references, or would you recommend a different design?

I did try to implement this back then, but at that time I didn't have `GuadConditions` yet to trigger the WaitSet. Would have to look back into my notes and at the code to give a definite answer here, but yes I think referencing the node would be needed. I like to check how `rclcpp` and `rclpy`  deal with this as well. (and maybe `rclrust` as well as they need to "document" object lifetimes explicitly)

> Perhaps RCLdotnet could just check if the Waitables have been disposed of as part of its SpinOnce routine.

This is a good idea worth considering as it would make things way easier 👍

Reply:
`use_sim_time` changes are made as a response to a service call, so the `WaitSet` must be active in the current implementation I have. That being said, it occurs in the services section of the spin which is after all of the subscriptions have already completed their callbacks. I'll need to read the `WaitSet` docs again on the RCL side to work out if that matters.

Reply:
The docs for `rcl_subscription_fini` describe the behaviour of `rcl_wait` [here](https://github.com/ros2/rcl/blob/bf5d4bcd85240068626630c78865e3e28ccce1b0/rcl/include/rcl/subscription.h#L174). As the subscription being disposed would be in response to an already completed `rcl_wait`, there shouldn't be an issue there. Should a subscription with messages ready be finalised as a response to an earlier subscription, the resulting take would fail - which is already handled in `RCLDotnet.Take`.

Reply:
> `use_sim_time` changes are made as a response to a service call, so the `WaitSet` must be active in the current implementation I have. That being said, it occurs in the services section of the spin which is after all of the subscriptions have already completed their callbacks. I'll need to read the `WaitSet` docs again on the RCL side to work out if that matters.

I don't think this depends on the subscriptions in the `WaitSet`. With an active `WaitSet` I meant that the `rcl_wait` is currently called and blocking. Given that we want to deactivate the Timer in an internal service callback and we are single-threaded right now, changing this to internal would support the internal use for sim-time. The rest can be done in another PR.